homeassistant/components/israel_rail/coordinator.py

```diff
@@ -0,0 +1,113 @@
+"""DataUpdateCoordinator for the israel rail integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from israelrailapi import TrainSchedule
+from israelrailapi.api import TrainRoute
+from israelrailapi.train_station import station_name_to_id
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from .const import DEFAULT_SCAN_INTERVAL, DEPARTURES_COUNT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class DataConnection:
+    """A connection data class."""
+
+    departure: datetime | None
+    duration: int | None
+    platform: str
+    remaining_time: str
+    start: str
+    destination: str
+    train_number: str
+    transfers: int
+
+
+def calculate_duration_in_seconds(start_time: str, end_time: str) -> int | None:
```

Comment:
We don't allow calculating state. Sensors should just return the native values of the API as appropriate according to the Home Assistant entity design guidelines.

Reply:
Removed